src/stream/src/executor/troublemaker.rs

```diff
@@ -0,0 +1,191 @@
+                        ) {
+                            vars.met_delete_before = true;
+                        }
+
+                        for (op, row) in
+                            make_some_trouble(&data_types, record, vars.met_delete_before)
+                        {
+                            if let Some(chunk) = vars.chunk_builder.append_row(op, row) {
+                                yield Message::Chunk(chunk);
+                            }
+                        }
+                    }
+
+                    if let Some(chunk) = vars.chunk_builder.take() {
+                        yield Message::Chunk(chunk);
+                    }
+                }
+                Message::Barrier(barrier) => {
+                    assert!(vars.chunk_builder.take().is_none(), "we don't merge chunks");
+                    yield Message::Barrier(barrier);
+                }
+                _ => yield msg,
+            }
+        }
+    }
+}
+
+impl Execute for TroublemakerExecutor {
+    fn execute(self: Box<Self>) -> BoxedMessageStream {
+        self.execute_inner().boxed()
+    }
+}
+
+fn make_some_trouble(
+    data_types: &[DataType],
+    record: Record<impl Row>,
+    can_change_op: bool,
+) -> impl Iterator<Item = (Op, OwnedRow)> + '_ {
+    let record = if can_change_op && rand::thread_rng().gen_bool(0.5) {
+        // Change the `Op`
+        match record {
+            Record::Insert { new_row } => Record::Delete {
+                old_row: new_row.into_owned_row(),
+            },
+            Record::Delete { old_row } => Record::Insert {
+                new_row: old_row.into_owned_row(),
+            },
+            Record::Update { old_row, new_row } => Record::Update {
+                old_row: new_row.into_owned_row(),
+                new_row: old_row.into_owned_row(),
+            },
+        }
+    } else {
+        // Just convert the rows to owned rows, without changing the `Op`
+        match record {
+            Record::Insert { new_row } => Record::Insert {
+                new_row: new_row.into_owned_row(),
+            },
+            Record::Delete { old_row } => Record::Delete {
+                old_row: old_row.into_owned_row(),
+            },
+            Record::Update { old_row, new_row } => Record::Update {
+                old_row: old_row.into_owned_row(),
+                new_row: new_row.into_owned_row(),
+            },
+        }
+    };
+
+    record
+        .into_rows()
+        .map(|(op, row)| (op, randomize_row(data_types, row)))
+}
+
+fn randomize_row(data_types: &[DataType], row: OwnedRow) -> OwnedRow {
+    let mut data = row.into_inner();
+
+    for (datum, data_type) in data.iter_mut().zip_eq_fast(data_types) {
+        match rand::thread_rng().gen_range(0..4) {
+            0 | 1 => {
+                // don't change the value
+            }
+            2 => {
+                *datum = None;
+            }
+            3 => {
+                *datum = match data_type {
```

Comment:
nit: We already have a random generator here: https://github.com/risingwavelabs/risingwave/blob/4f2f65e2b665b9eaa48720b2238222dfb73cc3da/src/common/src/field_generator/mod.rs#L43-L44

 I don't mind making a new one here as it's for testing purposes.

Reply:
Reused the existing one.